File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesKeywords.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.keyword.Keyword;
import dev.reuss.tmdb.common.keyword.KeywordsResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Keywords for a TMDB TV series.
 *
 * @param id      TMDB TV series id
 * @param results TV series keywords
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesKeywords(
        int id,
        List<Keyword> results
) implements KeywordsResponse, TmdbModel {

    public TvSeriesKeywords {
        results = TmdbCollections.list(results);
    }

    @Override
    public List<Keyword> keywords() {
        return results;
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesKeywords.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.keyword.Keyword
import dev.reuss.tmdb.common.keyword.KeywordsResponse

/**
 * Keywords for a TMDB TV series.
 *
 * @property id TMDB TV series id
 * @property keywords TV series keywords
 */
@JvmRecord
data class TvSeriesKeywords(
    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("results")
    override val keywords: List<Keyword> = emptyList()
) : KeywordsResponse, TmdbModel